### PR TITLE
R4 care team fhir document update

### PR DESCRIPTION
**Pull request by NimishaMN**
Description
----
Added CareTeam Resource to R4

Screenshot 2020-09-29 at 7 13 50 PM
Screenshot 2020-09-29 at 7 13 57 PM
Screenshot 2020-09-29 at 7 15 09 PM

PR Checklist
----
- [ ] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

**Comment by NimishaMN** [comment]
@NimishaMN Can you add the description and provide the screenshots of the local fhir.cerner.com. Also, Can you please add one more reviewer as you will need to have 2 persons approval for this to merge.

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [inline content/millennium/r4/care-provision/care-team.md:26]: Alignment

File: content/millennium/r4/care-provision/care-team.md
Comment (sonikrish) [inline content/millennium/r4/care-provision/care-team.md:26]: Remove extra space

Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md:26]: Role and Member are subparts of Participant. Hence to differentiate from other components added the spaces. Should the spaces be removed?

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:12]: I see that we have this sentence on CarePlan as well, but what information is it really communicating?

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:14]: Use regular single quote characters instead of tilted single quote characters

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [reply content/millennium/r4/care-provision/care-team.md:26]: the spaces should stay

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:32]: remove all instances of double blank lines

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:41]: there's no such element as CareTeam.activity, right?

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:62]: use the same bullet characters throughout

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:66]: the _id section says patient or encounter is needed, these sections say each is required

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:67]: add backticks to all parameter names

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:83]: only need to add extra to the Example name when there are multiple examples

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [inline content/millennium/r4/care-provision/care-team.md:43]: this implementation note seems applicable both to search and read. if it is, we'll probably want to move it up

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [inline lib/resources/example_json/r4_examples_care_team.rb:18]: use single quotes

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [inline lib/resources/example_json/r4_examples_care_team.rb:59]: don't need to wrap text this way in my opinion

File: lib/resources/r4/care_team.yaml
Comment (MaxPhilips) [inline lib/resources/r4/care_team.yaml:5]: patient doesn't appear to be needed in your .md

File: lib/resources/r4/care_team.yaml
Comment (MaxPhilips) [inline lib/resources/r4/care_team.yaml:17]: encounter doesn't appear to be needed in your .md

File: lib/resources/r4/care_team.yaml
Comment (MaxPhilips) [inline lib/resources/r4/care_team.yaml:31]: required and type keys do not appear to be needed in your .md - you are only adding a terminology table instead of a body fields table

File: lib/resources/r4/care_team.yaml
Comment (MaxPhilips) [inline lib/resources/r4/care_team.yaml:48]: missing final newline

**Review by MaxPhilips — CHANGES_REQUESTED** [review]

File: content/millennium/r4/care-provision/care-team.md
Comment (sonikrish) [reply content/millennium/r4/care-provision/care-team.md:38]: +1

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (sonikrish) [inline lib/resources/example_json/r4_examples_care_team.rb:15]: render the div with < > sign instead

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Removed.

File: lib/resources/r4/care_team.yaml
Comment (NimishaMN) [reply lib/resources/r4/care_team.yaml:50]: Encounter field is needed. Made the changes as per requirement.

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Addressed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/1c1aad456a43d4ba5ae6e7fc8620a05d19202987

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Removed https://github.com/cerner/fhir.cerner.com/pull/470/commits/1c1aad456a43d4ba5ae6e7fc8620a05d19202987

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Addressed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/1c1aad456a43d4ba5ae6e7fc8620a05d19202987

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Addressed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/1c1aad456a43d4ba5ae6e7fc8620a05d19202987

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Removed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/1c1aad456a43d4ba5ae6e7fc8620a05d19202987

File: lib/resources/r4/care_team.yaml
Comment (NimishaMN) [reply lib/resources/r4/care_team.yaml]: Removed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/1c1aad456a43d4ba5ae6e7fc8620a05d19202987

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Addressed https://github.com/cerner/fhir.cerner.com/pull/470/commits/59612acba6bd362fc38192acb7df99fd3a8db65c

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (NimishaMN) [reply lib/resources/example_json/r4_examples_care_team.rb]: Addressed https://github.com/cerner/fhir.cerner.com/pull/470/commits/6920a53fa4489aa0584dd2458c3d1517bbf629ff

File: lib/resources/r4/care_team.yaml
Comment (NimishaMN) [reply lib/resources/r4/care_team.yaml]: Added the required body fields and updated the terminology table. https://github.com/cerner/fhir.cerner.com/pull/470/commits/743b57306d7803dca473eb256c61982ec9c0ff62

File: lib/resources/r4/care_team.yaml
Comment (NimishaMN) [reply lib/resources/r4/care_team.yaml]: Added. https://github.com/cerner/fhir.cerner.com/pull/470/commits/743b57306d7803dca473eb256c61982ec9c0ff62

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (NimishaMN) [reply lib/resources/example_json/r4_examples_care_team.rb]: Addressed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/743b57306d7803dca473eb256c61982ec9c0ff62

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Removed Implementation Note section after discussion with @sonikrish . Came to a conclusion that Implementation Note section is not required as the text included, " The description of the CareTeam is included in the [CareTeam.text.div] element as ASCII text." doesn't serve the purpose of the section.
Commit: https://github.com/cerner/fhir.cerner.com/pull/470/commits/743b57306d7803dca473eb256c61982ec9c0ff62

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (NimishaMN) [reply lib/resources/example_json/r4_examples_care_team.rb]: Addressed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/1c1aad456a43d4ba5ae6e7fc8620a05d19202987#diff-05cdc79578a87a8b1066e2bca824be5f

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [reply content/millennium/r4/care-provision/care-team.md]: we want to keep this example, we should title it "Example" instead of "Example retrieve care team" though

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [reply content/millennium/r4/care-provision/care-team.md]: I think mentioning where the CareTeam description can be found is important to include, especially since CareTeam.description isn't a normal field on the API

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [reply lib/resources/example_json/r4_examples_care_team.rb]: with single quotes, the inner double quotes shouldn't need to be escaped

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [reply lib/resources/example_json/r4_examples_care_team.rb]: apply to all line wraps

File: lib/resources/r4/care_team.yaml
Comment (MaxPhilips) [reply lib/resources/r4/care_team.yaml:50]: this yaml file is consumed by various tables embedded in the markdown files. the only table I see in markdown is a terminology bindings table. since encounter is not bound to any terminology (as it is a reference field, not a coded field), I don't understand why it needs to be here

am i missing anything?

File: lib/resources/r4/care_team.yaml
Comment (MaxPhilips) [reply lib/resources/r4/care_team.yaml]: the way to address this comment is by removing required: and type: keys

those keys are needed when rendering body fields tables, but not needed when rendering terminology tables

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [inline lib/resources/example_json/r4_examples_care_team.rb:18]: does our code not put a space after Name: and Role: ? we'll need to fix that if so

Comment (sonikrish) [reply lib/resources/example_json/r4_examples_care_team.rb:18]: it doesn't

File: lib/resources/r4/care_team.yaml
Comment (sonikrish) [reply lib/resources/r4/care_team.yaml:50]: +1. Nimisha you can remove the fields which are not present in terminology bindings

File: content/millennium/r4/care-provision/care-team.md
Comment (sonikrish) [reply content/millennium/r4/care-provision/care-team.md]: But we don't have CareTeam.description supporting as of now not on the text div also. I think this got copied while looking through care plan since in CarePlan we do have a exception that description is only visible on the div part but not on the body. 
So, we should not have this under Implementation notes for CareTeam. Let me know if I am getting something wrong or missing?
Same applies for the CareTeam.activity we don't have any such field

File: lib/resources/r4/care_team.yaml
Comment (sonikrish) [reply lib/resources/r4/care_team.yaml]: +1

Comment (NimishaMN) [reply lib/resources/r4/care_team.yaml]: Addressed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/60de10bf8259905011fb638e99b4f1607031c126#diff-c37b2a12be12c28b23605fef2eacc3ceL17-L18

File: lib/resources/r4/care_team.yaml
Comment (NimishaMN) [reply lib/resources/r4/care_team.yaml]: Removed. https://github.com/cerner/fhir.cerner.com/pull/470/commits/60de10bf8259905011fb638e99b4f1607031c126#diff-c37b2a12be12c28b23605fef2eacc3ceL50-L62

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (NimishaMN) [reply lib/resources/example_json/r4_examples_care_team.rb]: Addressed https://github.com/cerner/fhir.cerner.com/pull/470/commits/31181ae509682c786080e6ef354f2c771f7a5bf2

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (NimishaMN) [reply lib/resources/example_json/r4_examples_care_team.rb]: Addressed https://github.com/cerner/fhir.cerner.com/pull/470/commits/31181ae509682c786080e6ef354f2c771f7a5bf2

File: content/millennium/r4/care-provision/care-team.md
Comment (NimishaMN) [reply content/millennium/r4/care-provision/care-team.md]: Addressed https://github.com/cerner/fhir.cerner.com/pull/470/commits/60de10bf8259905011fb638e99b4f1607031c126#diff-05cdc79578a87a8b1066e2bca824be5fR102

File: content/millennium/r4/care-provision/care-team.md
Comment (MaxPhilips) [reply content/millennium/r4/care-provision/care-team.md]: got it, yep, we'll need to remove this if we don't return the info in question

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [reply lib/resources/example_json/r4_examples_care_team.rb:18]: cool. then we'll need to update this example :)

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [reply lib/resources/example_json/r4_examples_care_team.rb]: i don't see a double quote fix in that commit. here's what they look like when they are escaped with backslashes:

Screen Shot 2020-10-07 at 1 37 50 PM

were you able to compile the site locally to check?

i pushed the fix with https://github.com/cerner/fhir.cerner.com/pull/470/commits/d035801ecfcbf522dfa51be6884f1011b582c086#diff-f7c10b53dd7ead0281cf4dbd1281a12cR15

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [reply lib/resources/example_json/r4_examples_care_team.rb]: i don't see a line wrapping change in that commit

i removed the line wrapping with https://github.com/cerner/fhir.cerner.com/pull/470/commits/d035801ecfcbf522dfa51be6884f1011b582c086#diff-f7c10b53dd7ead0281cf4dbd1281a12cR15 and fixed the lint error with https://github.com/cerner/fhir.cerner.com/pull/470/commits/1822bc1d106921d7c6662d7648c00bc10b79d01c#diff-f7c10b53dd7ead0281cf4dbd1281a12cR15

i also added a change to a DSTU 2 file that used the wrong rubocop namespace since I noticed it getting flagged while checking lint

File: lib/resources/example_json/r4_examples_care_team.rb
Comment (MaxPhilips) [reply lib/resources/example_json/r4_examples_care_team.rb]: updated https://github.com/cerner/fhir.cerner.com/pull/470/commits/d035801ecfcbf522dfa51be6884f1011b582c086#diff-f7c10b53dd7ead0281cf4dbd1281a12cR15

**Review by MaxPhilips — APPROVED** [review]

**Review by sonikrish — APPROVED** [review]

**Comment by MaxPhilips** [comment]
@NimishaMN since you've already attached verification screenshots, i am checking the box "Screenshot(s) of changes attached before changes merged."

**Comment by nishanthnamana** [comment]
**Validation from Live Site**

CareTeam_1
CareTeam_2
CareTeam_3
CareTeam_4

**Pull request merged**
